internal/pkg/v2/handler/core/data/device.go

```diff
@@ -0,0 +1,23 @@
+package data
+
+import (
+	"context"
+
+	"github.com/edgexfoundry/edgex-go/internal/core/data/config"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
+)
+
+func checkDevice(
+	device string,
+	ctx context.Context,
+	mdc metadata.DeviceClient,
+	configuration *config.ConfigurationStruct) error {
+
+	if configuration.Writable.MetaDataCheck {
+		_, err := mdc.CheckForDevice(ctx, device)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
Copy from v1 `internal/core/data/device.go`.

Reply:
add a comment states this function will be modified once V2 CheckDevice in core-metadata is available, and open an issue to track this after the PR opened